alacritty_terminal/src/term/mod.rs

```diff
@@ -1413,10 +1411,13 @@ impl<T> Term<T> {
         };
 
         // Cursor shape.
-        let hidden =
-            !self.mode.contains(TermMode::SHOW_CURSOR) || point.line >= self.screen_lines();
-        let cursor_shape = if hidden && !vi_mode {
-            point.line = Line(0);
+        let hidden = !show_cursor
+            || !self.mode.contains(TermMode::SHOW_CURSOR)
+            || point.line >= self.screen_lines();
+        let cursor_shape = if hidden {
+            if !vi_mode {
+                point.line = Line(0);
+            }
             CursorShape::Hidden
```

Comment:
This isn't correct though, is it? If `SHOW_CURSOR` isn't set, the vi mode cursor must not be hidden. That's at least one of the reason for the `!vi_mode` check and it looks like you break that here?